_posts/2025-02-10-announcing-ballerina-2201.11.0-swan-lake-update-11.md

```diff
@@ -0,0 +1,250 @@
+
+Performance tests were conducted to evaluate the heap memory usage of Ballerina Update 10 and Update 11. Since garbage collection causes fluctuations in memory usage during execution, this analysis focuses on the peak heap memory recorded for each test, demonstrating a significant reduction in memory consumption.  
+
+<img alt="Loops U10 Vs U11" src="/images/u11/peak-heap-memory-u10-vs-u11.png">
+
+| Scenario | U10 Average(MB) | U11 Average (MB)  | Improvement |
+| :---- | :---- | :---- | :---- |
+| HTTP Service with Workers | 3020.8 | 1945.6 | **35.6% decrease** |
+| HTTP Passthrough | 437 | 359 | **18.0% decrease** |
+| JSON to XML Service | 145.7 | 99.3 | **31.8% decrease** |
+
+### Strand Dump Tool
+
+The strand dump tool has been enhanced to support virtual threads. The report now includes the total strand count, identifies strands running in parallel and sequentially, and provides their corresponding stack traces.
+
+In the updated concurrency model, every strand is directly mapped to a Java virtual thread, and the tool extracts strand-related details from the virtual thread dump. However, since the thread dump does not include virtual thread states, the current version of the strand dump report does not display strand states.
+
+### New Runtime APIs
+
+The Ballerina Runtime Java APIs have been redesigned to enhance seamless integration for Java-to-Ballerina interoperability.   
+Here’s an example of how to invoke a Ballerina function from Java code.
+
+**Ballerina**
+
+```ballerina
+public function main() returns error? {
+   boolean b = callIsEven(2);
+}
+
+
+function isEven(int n) returns boolean {
+   return n % 2 == 0;
+}
+
+
+public isolated function callIsEven(int n) returns boolean = @java:Method {
+   'class: "org.ballerinalang.examples.Test",
+   name: "callIsEven"
+} external;
+```
+
+**Java**
+
+```java
+class Test {
+   public static boolean callIsEven(Environment env, long arg) {
+       return env.getRuntime().callFunction(new Module("testOrg", "foo"), "isEven", null, arg);
+   }
+}
+```
+
+The following code shows an example of calling an object method using Java API.
+
+**Ballerina**
+
+```ballerina
+import ballerina/jballerina.java;
+
+public class Person {
+
+
+   public string name;
+   public int age;
+  
+   function init(string name, int age) {
+       self.name = name;
+       self.age = age;
+   }
+
+
+   public function play(string sport = "cricket") returns string {
+       return sport;
+   }
+      
+   public function callPlayWithArgs(string s) returns string = @java:Method {
+       'class: "org.ballerinalang.examples.Test"
+   } external;
+}
+
+
+public function main() {
+   Person p = new Person("John", 30);
+   string sport = p.callPlayWithArgs("football");
+}
+
+```
+
+```java
+class Test {
+   public static BString callPlayWithArgs(Environment env, BObject object, BString bString) {
+       return env.getRuntime().callMethod(object, "play", null, bString);
+   }
+}
+```
+
+## Language updates
```

Comment:
@gimantha / @MaryamZi is it correct to list these under Language updates? As per the definition `data.` modules are part of Ballerina library right?

Reply:
As per offline chat with @gimantha we will move to library.